internal/server/server.go

```diff
@@ -68,8 +84,13 @@ func (s *Server) handleRoot() http.HandlerFunc {
 			return
 		}
 
-		// Make Subject claim the value used for login
-		login := claims.Subject
+		// possible values of Login claim are checked in cli beforehand
+		login := getValidClaim(claims, s.grafanaClaimsMap.Login)
+		name := getValidClaim(claims, s.grafanaClaimsMap.Name)
+		var email string
+		if claims.Email != "" {
+			email = claims.Email
+		}
```

Comment:
This seems like it can be set directly.  `claims.Email` is either an empty string or has value.

Reply:
done